src/yb/master/catalog_entity_info.h

```diff
@@ -482,7 +482,7 @@ class TableInfo : public RefCountedThreadSafe<TableInfo>,
   // Sorted index of tablet start partition-keys to TabletInfo.
   // The TabletInfo objects are owned by the CatalogManager.
   typedef std::map<std::string, TabletInfo *> TabletInfoMap;
-  TabletInfoMap tablet_map_;
+  TabletInfoMap tablet_map_ GUARDED_BY(lock_);
```

Comment:
This annotation and the one on `pending_tasks_` are both good additions, but unfortunately lead to some compiler errors under certain build circumstances.  This is caused by a failure of some compilers to recognize `shared_lock` as fulfilling the mutex-hold requirement.

Fortunately, there's an easy fix for this.  Yugabyte has a custom `SharedLock` implementation (`src/yb/util/shared_lock.h`) with a workaround for the compiler issue.  If you replace all instances of `shared_lock` in `catalog_entity_info.cc` with `SharedLock`, that should resolve the issue.